types/pdfmake/interfaces.d.ts

```diff
@@ -108,7 +108,7 @@ export type TableCell =
 
 export interface Table {
     body: TableCell[][];
-    widths?: Size[];
+    widths?: string | Size[];
```

Comment:
Can you provide a reference? I have not been able to find anything corroborating this.

Reply:
I could not find it in the documentation either, but here is the [related code section](https://github.com/bpampuch/pdfmake/blob/35f3ea19918f7a3b04aa54ad25e94db0c1bd5dac/src/DocMeasure.js#L634)

Reply:
I don't know, it looks to me this was done to support the dynamic nature of JS and the default 'auto' value, set right before. If we could use Size | Size[], I'd be more for it, but numbers are not allowed.

In the [playground](http://pdfmake.org/playground.html)
```javascript
dd = { content: { table: {
  widths: 50,
  body: [
    ['Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae'],
    ['Sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae']
  ]
}}};
```
gives an error and
```javascript
dd = { content: { table: {
  widths: '50',
  body: [
    ['Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae'],
    ['Sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae']
  ]
}}};
```
looks buggy.

Percentages, `'*'` and `'auto'` work fine. Any other string seem to give errors.

What do you think?

I think it'd be better to wait until pdmake supports Size, to avoid errors.

Reply:
As you mentioned '*' and 'auto' works fine, so having widths bound to `Size[]` type is limiting. 

I agree that `string | Size[]` is not restrictive enough.... We could do something like this instead:
```
widths?:  '*' | 'auto' | Size[];
```

Reply:
Sounds like a good compromise.